cmd/nerdctl/stats.go

```diff
@@ -465,8 +468,10 @@ func collect(cmd *cobra.Command, s *statsutil.Stats, waitFirst *sync.WaitGroup,
 			}
 		case err := <-u:
 			if err != nil {
-				s.SetError(err)
-				continue
+				if !strings.Contains(err.Error(), "no running task found") {
+					s.SetError(err)
+					continue
+				}
```

Comment:
this should not happen, because container are removed once received the delete event: https://github.com/containerd/nerdctl/blob/master/cmd/nerdctl/stats.go#L229-L245

Reply:
Nope, the origin code is not cover the existing exited container if we do not handle this, there will be infinite loop here

Reply:
in this case we should handle exist event https://pkg.go.dev/github.com/containerd/containerd/api/events#TaskExit 
May be pause event too

Reply:
we are dealing with container fetched through containerd API,  I believe that we can catch these events

Reply:
- Already exited container should not be added to the `cStats ` 
- When container exits it should be removed from `cStats` using the callback event

Reply:
> * Already exited container should not be added to the `cStats`

this is a little bit different with Docker, Docker will list all exited container metric with nil or 0 in `docker stats -a`

Reply:
I don't see any benefit behind leaving nil entries. We know in advance that exited containers have nil entry values

Reply:
@AkihiroSuda @fahedouch

Reply:
I am ok with being compatible with docker and displaying entries with nil values for exited containers and without displaying  error message `no running task found`

Reply:
Docker-compatible behavior is preferable, but if it is difficult we can just document it and call it a day